sonar-project.properties

```diff
@@ -0,0 +1,62 @@
+sonar.projectKey=unreal-plugin
+sonar.projectName=UnrealPlugin
+sonar.sources=.
+sonar.cplusplus.file.suffixes=.cpp,.h,.hpp
+sonar.cpp.file.suffixes=.cpp,.h,.hpp
+
+# Exclude build directories and third-party code
+sonar.exclusions=**/Build/**,**/Intermediate/**,**/ThirdParty/**,**/Content/**
+
+# Configure C++ analysis
+sonar.cpp.errorRecoveryEnabled=true
+sonar.cpp.microsoft=true
+sonar.cpp.standardLibrary=microsoft
+
+# Ignore Unreal-specific naming conventions
+sonar.issue.ignore.multicriteria=e1,e2,e3,e4,e5,e6
+
+# Custom parameters for Unreal-specific patterns
+sonar.cplusplus.complexity.threshold=15
+sonar.cplusplus.lines.of.code.per.file.threshold=1000
+sonar.cplusplus.maximum.function.parameters.threshold=8
+
+# Allow UPROPERTY macro usage
```

Comment:
May I ask why do we choose these rule key? Any guideline we followed?

Reply:
It's a good call out - I really don't think we need these at all.  Removed this and several other parts I'm not yet sure we need.